src/Core_Command.php

```diff
@@ -116,6 +116,12 @@ public function check_update( $_, $assoc_args ) {
 	 * [--version=<version>]
 	 * : Select which version you want to download. Accepts a version number, 'latest' or 'nightly'.
 	 *
+	 * [--retry=<retry>]
+	 * : Number of retries in case of failure. Default is 0.
+	 *
+	 * [--retry-delay=<retry-delay>]
+	 * : Delay between retries in seconds. Default is 5 seconds.
+	 *
```

Comment:
Given our conversation on Zoom today, I think we should change this slightly.

* Within `wp core download`, I think we should only have `--retry=<count>` as an argument. This will let us be opinionated about exponential backoff.
* We can update `Utils\http_request()` to support `'retry'` and `'retry_delay'` as arguments.
* The `--retry=<count>` argument should be applied to all HTTP requests within `wp core download`.

If WordPress.org is down in some manner, it's not helpful for WP-CLI users to keep hammering it with requests. Their systems should be tolerant to this fault. However, `--retry=<count>` with exponential (or linear) backoff (5 to ~30 seconds) and a maximum count seems like a good compromise.

We should have some tests!

Reply:
@danielbachhuber As discussed in the call, we decided to wrap the whole try-catch block inside the do-while loop to handle the retry feature. But after taking a deeper look into it there are many early and exception throwing that may break the retry loop in between. 

We should keep the retry loop in the `download` command only because updating the `http_request` util will affect other commands as well.